From 49c5069a2f48b0873b6c7a4f3acb63aa8d747e44 Mon Sep 17 00:00:00 2001
From: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
Date: Tue, 22 May 2018 15:22:42 +0900
Subject: [PATCH 403/909] iommu/ipmmu-vmsa: Fix IMCTR_RTSEL_MASK for R-Car Gen3
 and Gen2

commit 29cc0ed66fdc3b89a997055621ac59bada16802a from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch adds IMCTR_RTSEL_MASK_GEN3 for R-Car Gen3 and changes
IMCTR_RTSEL_MASK to IMCTR_RTSEL_MASK_GEN2.

Signed-off-by: Kazuya Mizuguchi <kazuya.mizuguchi.ks@renesas.com>
Signed-off-by: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/iommu/ipmmu-vmsa.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/iommu/ipmmu-vmsa.c b/drivers/iommu/ipmmu-vmsa.c
index 9717a048f3be..6690d180e6c9 100644
--- a/drivers/iommu/ipmmu-vmsa.c
+++ b/drivers/iommu/ipmmu-vmsa.c
@@ -123,7 +123,8 @@ static struct ipmmu_vmsa_device *to_ipmmu(struct device *dev)
 #define IMCTR				0x0000
 #define IMCTR_TRE			(1 << 17)
 #define IMCTR_AFE			(1 << 16)
-#define IMCTR_RTSEL_MASK		(3 << 4)
+#define IMCTR_RTSEL_MASK_GEN2		(3 << 4)
+#define IMCTR_RTSEL_MASK_GEN3		(7 << 4)
 #define IMCTR_RTSEL_SHIFT		4
 #define IMCTR_TREN			(1 << 3)
 #define IMCTR_INTEN			(1 << 2)
-- 
2.17.1

